src/lib/dialog_ui.sh

```diff
@@ -241,6 +241,78 @@ function create_loading_screen_notification()
   run_dialog_command "$cmd" "$flag"
 }
 
+function spinFrame()
+{
+  local LC_CTYPE=C
+  local spin='⣾⣽⣻⢿⡿⣟⣯⣷'
+  local charwidth=3
+
+  local i=$1
+  i=$(((i + charwidth) % ${#spin}))
+  printf "%s" "${spin:$i:$charwidth}"
+}
+
+# Create simple async loading screen notification for delayed actions.
+#
+# The caller is required to use stop_loading() with the pid of this function call
+# afterwards to stop the loading screen.
+#
+# @loading_message: The message to be displayed to the user while loading.
+# @height: Menu height in lines size, the default value is 8.
+# @width: Menu width in column size, the default value is 60.
+# @flag How to display a command, the default value is
+#   "SILENT". For more options see `src/lib/kwlib.sh` function `cmd_manager`
+#
+# Return:
+# There is no return. The function just displays the infobox and returns,
+# assuming the next command is going to delay its completionand that another
+# screen is going to clear the current (the infobox).
+function create_async_loading_screen_notification()
```

Comment:
@Andregnl and @fernandohjunqueira, this is about the below problem that happens when running the tests that we talked offline.

![photo_2023-10-31 01 04 53](https://github.com/kworkflow/kworkflow/assets/95055784/09df89ad-f54b-45d8-b7ec-150155c4d9a0)

The problem is that the test case `test_handle_bookmark_action` of the test file `tests/ui/patch_hub/patchset_details_and_actions_test.sh` tests the function `handle_bookmark_action`, which calls `create_async_loading_screen_notification` underneath, and makes this dialog screen appear without failing the test.

The way to solve this is by mocking the function inside the test case (like I showed you in the past) to avoid creating a dialog screen. This mock is a little more complicated, because `handle_bookmark_action` also calls `stop_loading`, so we need to make this process continuous. Below is the diff to ease your work

```bash
diff --git a/tests/ui/patch_hub/patchset_details_and_actions_test.sh b/tests/ui/patch_hub/patchset_details_and_actions_test.sh
index 0108dd1..12afc4d 100755
--- a/tests/ui/patch_hub/patchset_details_and_actions_test.sh
+++ b/tests/ui/patch_hub/patchset_details_and_actions_test.sh
@@ -154,9 +154,11 @@ test_handle_bookmark_action()
   local output
 
   # shellcheck disable=SC2317
-  function create_loading_screen_notification()
+  function create_async_loading_screen_notification()
   {
-    return
+    while true; do
+      sleep 0.1
+    done
   }
 
   # We need to mock download_series or else the test
```

Ping me if you don't understand my explanation 😬